common/messaging/kafkaClient.go

```diff
@@ -65,7 +66,11 @@ var _ Client = (*kafkaClient)(nil)
 // NewKafkaClient is used to create an instance of KafkaClient
 func NewKafkaClient(kc *KafkaConfig, metricsClient metrics.Client, zLogger *zap.Logger, logger log.Logger, metricScope tally.Scope,
 	checkCluster, checkApp bool) Client {
-	kc.Validate(checkCluster, checkApp)
+	err := kc.Validate(checkCluster, checkApp)
+	if err != nil {
+		logger.Warn("Wrong Kafka config.", tag.Error(err))
+		return nil
```

Comment:
return nil will cause panic if kafka client is still used

worth thinking about not initialize kafka client if not used